public/components/common/modules/discover/discover.tsx

```diff
@@ -108,7 +108,7 @@ export const Discover = compose(
   constructor(props) {
     super(props);
     this.KibanaServices = getDataPlugin();
-    this.filterManager = props.shareFilterManager ? this.KibanaServices.query.filterManager : new FilterManager(getUiSettings);
+    this.filterManager = props.shareFilterManager ? this.KibanaServices.query.filterManager : this.KibanaServices.query.filterManager;
```

Comment:
Here, `this.filterManager` always has the same value, don't?

Why not, simplify to:
```
this.filterManager = this.Kibanaservices.query.filerManager
```